packages/server/graphql/private/mutations/processRecurrence.ts

```diff
@@ -0,0 +1,119 @@
+import {MeetingSeries} from '../../../postgres/queries/getMeetingSeriesByIds'
+import {analytics} from '../../../utils/analytics/analytics'
+import publish from '../../../utils/publish'
+import standardError from '../../../utils/standardError'
+import createTeamPrompt from '../../mutations/helpers/createTeamPrompt'
+import endTeamPrompt from '../../mutations/helpers/endTeamPrompt'
+import isStartMeetingLocked from '../../mutations/helpers/isStartMeetingLocked'
+import {IntegrationNotifier} from '../../mutations/helpers/notifications/IntegrationNotifier'
+import {MutationResolvers} from '../resolverTypes'
+
+const startRecurringTeamPrompt = async (
+  meetingSeries: MeetingSeries,
+  startTime: Date,
+  dataLoader,
+  r,
+  subOptions
+) => {
+  const {teamId, facilitatorId} = meetingSeries
+
+  // AUTH
+  const unpaidError = await isStartMeetingLocked(teamId, dataLoader)
+  if (unpaidError) return standardError(new Error(unpaidError), {userId: facilitatorId})
+
+  const meeting = await createTeamPrompt(teamId, facilitatorId, r, dataLoader)
+
+  const formattedDate = new Date().toLocaleDateString('en-US', {
+    month: 'short',
+    day: 'numeric'
+  })
+
+  meeting.name = `${meetingSeries.title} - ${formattedDate}`
+  meeting.scheduledEndTime = new Date(startTime.getTime() + ms(`${meetingSeries.duration}m`))
+  meeting.meetingSeriesId = meetingSeries.id
+
+  await r.table('NewMeeting').insert(meeting).run()
+
+  IntegrationNotifier.startMeeting(dataLoader, meeting.id, teamId)
+  analytics.meetingStarted(facilitatorId, meeting, undefined, true)
+  const data = {teamId, meetingId: meeting.id}
+  publish(SubscriptionChannel.TEAM, teamId, 'StartTeamPromptSuccess', data, subOptions)
+}
+
+const processRecurrence: MutationResolvers['processRecurrence'] = async (
+  _source,
+  {},
+  {authToken, dataLoader, socketId: mutatorId}
+) => {
+  const r = await getRethink()
+  const now = new Date()
+  const operationId = dataLoader.share()
+  const subOptions = {mutatorId, operationId}
+
+  // VALIDATION
+
+  // RESOLUTION
+  // Find any meetings with a scheduledEndTime before now, and close them
+  const teamPromptMeetingsToEnd = (await r
+    .table('NewMeeting')
+    .filter({meetingType: 'teamPrompt'})
+    .filter((row) => row.hasFields('endedAt').not())
+    .filter((row) => row('scheduledEndTime').lt(now))
+    .run()) as MeetingTeamPrompt[]
+
+  let meetingsEnded = 0
+  await Promise.all(
+    teamPromptMeetingsToEnd.map(async (meeting) => {
+      const res = await endTeamPrompt({meeting, now, dataLoader, r, subOptions})
+      if (!('error' in res)) meetingsEnded++
+    })
+  )
+
+  let meetingsStarted = 0
+
+  // For each active meeting series, get the meeting start times (according to rrule) after the most
+  // recent meeting start time and before now.
+  const activeMeetingSeries = await getActiveMeetingSeriesQuery.run(undefined, getPg())
```

Comment:
+1 as a good practice, I'd still create a wrapper method for this just in case we'd need to map results and prevent direct use of `getPg` in mutations (just like for other queries)